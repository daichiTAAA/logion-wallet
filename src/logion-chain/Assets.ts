import { ApiPromise } from '@polkadot/api';
import { StorageKey, Option } from '@polkadot/types';
import {
    AssetId as PolkadotAssetId,
    AssetDetails as PolkadotAsset,
    AssetMetadata as PolkadotAssetMetadata,
    Call
} from '@polkadot/types/interfaces';
import { randomBytes } from 'crypto';
import BN from 'bn.js';
import {
    AssetId as AssetIdType,
    AssetBalance as AssetBalanceType,
    AssetMetadata as AssetMetadataType,
    Asset as AssetType,
    AssetWithBalance as AssetWithBalanceType,
} from './Types';

import {
    SignAndSendCallback,
    ErrorCallback,
    signAndSend,
    Unsubscriber
} from './Signature';

export const DEFAULT_ASSETS_DECIMALS = 18;

export interface AssetCreationParameters {
    signerId: string,
    callback: SignAndSendCallback,
    errorCallback: ErrorCallback,
    api: ApiPromise,
}

export type AssetId = AssetIdType;

export interface AssetCreationResult {
    assetId: AssetId,
    unsubscriber: Unsubscriber,
}

export async function createAsset(parameters: AssetCreationParameters): Promise<AssetCreationResult> {
    const {
        api,
        signerId,
        callback,
        errorCallback,
    } = parameters;

    const assetId = await nextAssetId(api);
    const assetCreate: any = api.tx.assets.create; // circumvent typing in order to add missing argument
    const unsubscriber = signAndSend({
        signerId,
        submittable: assetCreate(assetId, signerId, 10, 1),
        callback,
        errorCallback,
    });

    return {
        assetId,
        unsubscriber
    };
}

async function nextAssetId(api: ApiPromise): Promise<AssetId> {
    const BYTES_IN_ASSET_ID = 64 / 8;
    const value = randomBytes(BYTES_IN_ASSET_ID);
    let candidate = new BN(value.toString('hex'), 16);
    let candidateUsed = await assetExists(api, candidate);

    let iteration = 0;
    const MAX_ITERATIONS = 10;
    while(candidateUsed && iteration < MAX_ITERATIONS) {
        candidate = new BN(value.toString('hex'), 16);
        candidateUsed = await assetExists(api, candidate);
        ++iteration;
    }

    if(candidateUsed) {
        throw new Error("Unable to select an unused ID");
    } else {
        return candidate;
    }
}

async function assetExists(api: ApiPromise, id: AssetId): Promise<boolean> {
    const asset = await api.query.assets.asset(id);
    return asset.isSome;
}

export type AssetMetadata = AssetMetadataType;

export interface SetAssetMetadataParameters {
    signerId: string,
    callback: SignAndSendCallback,
    errorCallback: ErrorCallback,
    api: ApiPromise,
    assetId: AssetId,
    metadata: AssetMetadata,
}

export function setAssetMetadata(parameters: SetAssetMetadataParameters): Unsubscriber {
    const {
        api,
        signerId,
        callback,
        errorCallback,
        assetId,
        metadata,
    } = parameters;
    return signAndSend({
        signerId,
        submittable: api.tx.assets.setMetadata(assetId, metadata.name, metadata.symbol, metadata.decimals),
        callback,
        errorCallback,
    });
}
export type AssetBalance = AssetBalanceType;

export function balanceFromAmount(tokens: number, decimals: number): AssetBalance {
    let tokensBN = new BN(tokens);
    let decimalsBN = new BN(decimals);
    let ten = new BN(10);
    return tokensBN.mul(ten.pow(decimalsBN));
}

export interface MintParameters {
    signerId: string,
    beneficiary: string,
    callback: SignAndSendCallback,
    errorCallback: ErrorCallback,
    api: ApiPromise,
    assetId: AssetId,
    amount: AssetBalance
}

export function mintTokens(parameters: MintParameters): Unsubscriber {
    const {
        api,
        signerId,
        beneficiary,
        callback,
        errorCallback,
        assetId,
        amount,
    } = parameters;
    return signAndSend({
        signerId,
        submittable: api.tx.assets.mint(assetId, beneficiary, amount),
        callback,
        errorCallback,
    });
}

export function tokensFromBalance(balance: AssetBalance, decimals: number): string {
    let balanceBN = new BN(balance);
    let exponent = new BN(decimals);
    let ten = new BN(10);
    return balanceBN.div(ten.pow(exponent)).toString();
}

export interface AssetBalanceParameters {
    api: ApiPromise,
    assetId: AssetId,
    decimals: number,
    address: string,
}

export async function assetBalance(parameters: AssetBalanceParameters): Promise<string> {
    const {
        api,
        assetId,
        decimals,
        address,
    } = parameters;
    const balance = await api.query.assets.account(assetId, address);
    return tokensFromBalance(balance.balance, decimals);
}

export interface GetAssetsParameters {
    api: ApiPromise,
}

export type Asset = AssetType;

export async function getAssets(parameters: GetAssetsParameters): Promise<Asset[]> {
    const {
        api,
    } = parameters;
    const assetIds = extractAssetIds(await api.query.assets.asset.keys());
    const assetDetails: Option<PolkadotAsset>[] = await Promise.all(assetIds.map(assetId => api.query.assets.asset(assetId)));
    const metadata: PolkadotAssetMetadata[] = await Promise.all(assetIds.map(assetId => api.query.assets.metadata(assetId)));
    return assetIds.map((assetId, index) => {
        return {
            assetId,
            issuer: assetDetails[0].unwrap().issuer.toString(),
            metadata: {
                name: metadata[index].name.toUtf8(),
                symbol: metadata[index].symbol.toUtf8(),
                decimals: metadata[index].decimals.toNumber(),
            }
        };
    });
}

function extractAssetIds(keys: StorageKey<[PolkadotAssetId]>[]): AssetId[] {
    return keys.map(({ args: [assetId] }) => assetId);
}

export interface AccountBalanceParameters {
    api: ApiPromise,
    accountId: string,
    assets?: Asset[],
}

export type AssetWithBalance = AssetWithBalanceType;

export async function accountBalance(parameters: AccountBalanceParameters): Promise<AssetWithBalance[]> {
    const {
        api,
        accountId,
        assets,
    } = parameters;
    

    let theAssets;
    if(assets !== undefined) {
        theAssets = assets;
    } else {
        theAssets = await getAssets({api: api!});
    }
    const balances: string[] = await Promise.all(theAssets.map(asset => assetBalance({
        api,
        assetId: asset.assetId,
        decimals: asset.metadata.decimals,
        address: accountId
    })));
    return theAssets.map((asset, index) => ({
        asset,
        balance: balances[index],
    }));
}

export interface BuildTransferCallParameters {
    api: ApiPromise,
    assetId: AssetId,
    target: string,
    amount: AssetBalance,
}

export function buildTransferCall(parameters: BuildTransferCallParameters): Call {
    const {
        api,
        assetId,
        target,
        amount,
    } = parameters;

    return api.createType('Call', api.tx.assets.transfer(assetId, target, amount));
}
