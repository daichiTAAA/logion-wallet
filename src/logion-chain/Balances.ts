import { ApiPromise } from '@polkadot/api';
import { PrefixedNumber, ScientificNumber, convertToPrefixed, NONE, ATTO } from "./numbers";
import { ExtrinsicSubmissionParameters, Unsubscriber, signAndSend } from './Signature';
import { Call } from "@polkadot/types/interfaces";
import { SubmittableExtrinsic } from "@polkadot/api/submittable/types";

const LGNT_DECIMALS = 18;
export const LGNT_SMALLEST_UNIT = ATTO;
export const SYMBOL = "LGNT";

export interface GetAccountDataParameters {
    api: ApiPromise,
    accountId: string,
}

export interface AccountData {
    available: string,
    reserved: string,
    total: string,
}

export async function getAccountData(parameters: GetAccountDataParameters): Promise<AccountData> {
    const {
        api,
        accountId,
    } = parameters;

    const accountData = await api.query.system.account(accountId);

    return {
        available: accountData.data.free.toString(),
        reserved: accountData.data.reserved.toString(),
        total: accountData.data.free.add(accountData.data.reserved).toString(),
    };
}
export interface Coin {
    id: string,
    name: string,
    iconId: string,
    iconType: 'svg' | 'png',
    symbol: string,
}

export interface CoinBalance {
    coin: Coin,
    balance: PrefixedNumber,
    level: number,
}

export async function getBalances(parameters: GetAccountDataParameters): Promise<CoinBalance[]> {
    const {
        api,
        accountId,
    } = parameters;

    const data = await getAccountData({
        api: api!,
        accountId,
    });

    const logAvailable = scientificLogBalance(data.total).optimizeScale(3);
    const logPrefixedAvailable = convertToPrefixed(logAvailable);
    const logLevel = logAvailable.divideBy(ARTIFICIAL_MAX_BALANCE).toNumber();

    return [
        buildCoinBalance('lgnt', logPrefixedAvailable, logLevel),
        buildCoinBalance('dot', new PrefixedNumber("0", NONE), 1)
    ];
}

export function scientificLogBalance(tokens: string): ScientificNumber {
    return new ScientificNumber(tokens, -LGNT_DECIMALS).optimizeScale(3);
}

export function prefixedLogBalance(tokens: string): PrefixedNumber {
    const scientific = scientificLogBalance(tokens);
    return convertToPrefixed(scientific);
}

const ARTIFICIAL_MAX_BALANCE = scientificLogBalance("100");

function buildCoinBalance(coinId: string, balance: PrefixedNumber, level: number): CoinBalance {
    const coin = getCoin(coinId);
    return {
        coin,
        balance,
        level,
    }
}

export function getCoin(coinId: string): Coin {
    if(coinId === 'dot') {
        return {
            id: 'dot',
            name: 'Polkadot',
            iconId: 'dot',
            iconType: 'png',
            symbol: 'DOT',
        };
    } else if(coinId === "lgnt") {
        return {
            id: 'lgnt',
            name: 'logion Token',
            iconId: 'lgnt',
            iconType: 'svg',
            symbol: SYMBOL,
        };
    } else {
        throw new Error(`Unsupported coin ${coinId}`);
    }
}

export interface TransferParameters extends ExtrinsicSubmissionParameters, BuildTransferCallParameters {
}

export function transfer(parameters: TransferParameters): Unsubscriber {
    const {
        signerId,
        callback,
        errorCallback,
    } = parameters;

    return signAndSend({
        signerId,
        submittable: transferSubmittable(parameters),
        callback,
        errorCallback,
    });
}

export interface BuildTransferCallParameters {
    api: ApiPromise;
    destination: string;
    amount: PrefixedNumber;
}

function transferSubmittable(parameters: BuildTransferCallParameters): SubmittableExtrinsic<'promise'> {
    const {
        api,
        destination,
        amount
    } = parameters;
    return api.tx.balances.transfer(destination, amount.convertTo(LGNT_SMALLEST_UNIT).coefficient.unnormalize())
}

export function buildTransferCall(parameters: BuildTransferCallParameters): Call {
    return parameters.api.createType('Call', transferSubmittable(parameters))
}

export async function estimateFee(parameters: BuildTransferCallParameters): Promise<PrefixedNumber> {
    const submittable = transferSubmittable(parameters);
    const paymentInfo = await submittable.paymentInfo(parameters.destination);
    return new PrefixedNumber(paymentInfo.partialFee.toString(), LGNT_SMALLEST_UNIT);
}
