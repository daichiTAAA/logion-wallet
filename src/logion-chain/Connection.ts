import config from '../config';

export interface NodeMetadata {
    peerId: string
}

export function getEndpoints(): string[] {
    const providerSocket = config.PROVIDER_SOCKET;
    if(providerSocket !== undefined) {
        return [ providerSocket ];
    } else {
        const sockets = config.edgeNodes.map(node => node.socket);
        sockets.sort(() => Math.random() - 0.5);
        return sockets;
    }
}
