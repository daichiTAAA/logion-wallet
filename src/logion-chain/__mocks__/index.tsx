import { Extension } from '../Keys';

const LogionChainContextProvider = (props: any) => null;

let context = {
    apiState: 'CONNECT_INIT',
    injectedAccounts: []
};

export function setContextMock(value: any) {
    context = value;
}

export function useLogionChain() {
    return context;
}

let extensionAvailable = false;

export function setExtensionAvailable(value: boolean) {
    extensionAvailable = value;
}

export function isExtensionAvailable() {
    return extensionAvailable;
}

let recommendedExtensionValue: Extension | null = null;

export function setRecommendedExtension(value: Extension | null) {
    recommendedExtensionValue = value;
}

export function recommendedExtension(): Extension | null {
    return recommendedExtensionValue;
}

export {
    LogionChainContextProvider,
};
