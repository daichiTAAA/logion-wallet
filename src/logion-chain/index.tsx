// Imports
import { KeyringPair as PolkadotKeyringPair } from '@polkadot/keyring/types';
import {
    Block as PolkadotBlock,
    Hash as PolkadotHash,
    Header as PolkadotHeader,
    SignedBlock as PolkadotSignedBlock,
    EventRecord as PolkadotEventRecord,
    Extrinsic as PolkadotExtrinsic,
} from '@polkadot/types/interfaces';
import keyring from '@polkadot/ui-keyring';
import { InjectedAccountWithMeta as PolkadotInjectedAccountWithMeta } from '@polkadot/extension-inject/types';
import { ISubmittableResult as PolkadotISubmittableResult } from '@polkadot/types/types';

import {
    LogionChainContextProvider,
    useLogionChain,

    ExtrinsicFetchSpecification as ContextExtrinsicFetchSpecification,
    ApiState as ContextApiState,
    NodeMetadata as ContextNodeMetadata,
} from './LogionChainContext';

import {
    signAndSend,
    replaceUnsubscriber,
    sign,

    ExtrinsicSignatureParameters as SignatureExtrinsicSignatureParameters,
    Unsubscriber as SignatureUnsubscriber,
    SignAndSendCallback as SignatureSignAndSendCallback,
    AttributesSignatureParameters as SignatureStringSignatureParameters,
} from './Signature';

import {
    getTimestamp,
    fetchExtrinsics,
    hashEquals,

    ExtrinsicsAndHead as BlocksExtrinsicsAndHead,
} from './Blocks';

import {
    isExtensionAvailable,
    recommendedExtension
} from './Keys';

// Re-exports
export {
    useLogionChain,
    LogionChainContextProvider,
    getTimestamp,
    fetchExtrinsics,
    hashEquals,
    signAndSend,
    replaceUnsubscriber,
    isExtensionAvailable,
    recommendedExtension,
    keyring,
    sign,
};

export type Unsubscriber = SignatureUnsubscriber;
export type ExtrinsicSignatureParameters = SignatureExtrinsicSignatureParameters;
export type ExtrinsicFetchSpecification = ContextExtrinsicFetchSpecification;

export type ExtrinsicsAndHead = BlocksExtrinsicsAndHead;

export type KeyringPair = PolkadotKeyringPair;
export type Hash = PolkadotHash;
export type SignedBlock = PolkadotSignedBlock;
export type Header = PolkadotHeader;
export type Block = PolkadotBlock;
export type EventRecord = PolkadotEventRecord;
export type Extrinsic = PolkadotExtrinsic;
export type SignAndSendCallback = SignatureSignAndSendCallback;
export type ApiState = ContextApiState;
export type NodeMetadata = ContextNodeMetadata;
export type InjectedAccountWithMeta = PolkadotInjectedAccountWithMeta;
export type StringSignatureParameters = SignatureStringSignatureParameters;
export type ISubmittableResult = PolkadotISubmittableResult;
