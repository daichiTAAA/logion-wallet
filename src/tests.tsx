import { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import reactRenderer, { ReactTestRenderer, ReactTestRendererJSON, act as reactAct } from 'react-test-renderer';

export function render(element: ReactElement): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    const component = reactRenderer.create(element);
    return component.toJSON();
}

export function renderer(element: ReactElement): ReactTestRenderer {
    return reactRenderer.create(element);
}

export function shallowRender<E extends ReactElement>(element: ReactElement): E {
    const renderer = createRenderer();
    renderer.render(element);
    return renderer.getRenderOutput();
}

export function mockAccount(address: string, name: string) {
    return {
        address,
        meta: {
            name
        }
    };
}

export { reactAct as act };
