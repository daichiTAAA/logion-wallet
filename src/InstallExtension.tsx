 import React from 'react';
 import Jumbotron from 'react-bootstrap/Jumbotron';

import { recommendedExtension } from './logion-chain';
import Shell from './Shell';

export default function InstallExtension() {
    const extension = recommendedExtension();
    if(extension !== null) {
        return (
            <Shell>
                <Jumbotron>
                    <h1>Install the extension</h1>
                </Jumbotron>
                <p>Please install <a href={extension.url}>this browser extension</a> in order to store
                your accounts in a secure way. Once done, authorize the Logion wallet to access it
                and create or import an account.</p>
            </Shell>
        );
    } else {
        return (
            <Shell>
                <p>Your browser is currently not supported, please try with
                <a href="https://www.google.com/chrome/">Chrome</a> or
                <a href="https://www.mozilla.org/firefox/">Firefox</a></p>
            </Shell>
        );
    }
}
