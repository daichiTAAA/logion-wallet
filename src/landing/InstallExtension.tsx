import { recommendedExtension } from "../logion-chain/Keys";

export function InstallExtension() {
    const extension = recommendedExtension();
    if(extension !== null) {
        return (
            <>
                <h1>logion authentification system installation tutorial</h1>

                <p>To access logion legal protection services, you just have to create a Polkadot account.</p>

                <p>To do so, you need to install the Polkadot.js extension to your current browser (Firefox or Chrome are supported).</p>

                <p>This tutorial will help you to execute the installation of the authentification system used by logion.</p>

                <p>Each step is followed by a small animation that visually shows what you have to do, going through each related bullet point.</p>

                <h2>A - Find the polkadot.js extension in your browser Appstore</h2>

                <p>Go to the official polkadot.org extension website and select the right extension considering the browser you want to use while accessing the logion platform:<br/>
                <a href="https://polkadot.js.org/extension/" target="_blank" rel="noreferrer">https://polkadot.js.org/extension/</a></p>

                <img src={`${process.env.PUBLIC_URL}/assets/landing/extension-selection.png`} style={{maxWidth: "100%"}} alt="polkadot extension page"/>

                <h2>B - Polkadot extension installation</h2>

                <p>Install the polkadot.js extension to your browser by clicking on the related button (here for firefox but it’s exactly similar on Chrome):</p>

                <ul>
                    <li>Click on “Add to Firefox”</li>
                    <li>Click on “Add”</li>
                    <li>Click on “Okay”</li>
                </ul>

                <p>Finally:</p>

                <ul>
                    <li>Click on the Polkadot extension orange logo</li>
                    <li>Click on the “Understood, let me continue” orange button</li>
                </ul>

                <p>You are ready to create your first Polkadot account.</p>

                <img src={`${process.env.PUBLIC_URL}/assets/landing/logion-step0.gif`} style={{maxWidth: "100%"}} alt="step 0 animation"/>

                <h2>C - Polkadot account creation</h2>

                <p>To create a Polkadot account, please :</p>

                <ul>
                    <li>Click on the “Polkadot.js extension” orange logo</li>
                    <li>Select “Pop up” in the “Notification" dropdown menu</li>
                    <li>Select “open extension in new window” at the bottom of the extension: the extension will open itself in a new tab</li>
                    <li>Click anywhere in the underlying tab, outside of the Polkadot extension popup to make it disappeared</li>
                </ul>

                <img src={`${process.env.PUBLIC_URL}/assets/landing/logion-step1.gif`} style={{maxWidth: "100%"}} alt="step 1 animation"/>

                <ul>
                    <li>Click on the “+” button and then on “Create a new account”</li>
                    <li>Copy the 12 words seed phrase in a secure way (we strongly advise using a password manager such as <a href="https://keepassxc.org/" target="_blank" rel="noreferrer">KeePassXC</a> by clicking on “Copy to clipboard</li>
                    <li>Check the “I have saved my mnemonic seed safely” box</li>
                    <li>Click on the “Next Step” orange button</li>
                </ul>

                <img src={`${process.env.PUBLIC_URL}/assets/landing/logion-step1-5.gif`} style={{maxWidth: "100%"}} alt="step 1.5 animation"/>

                <p>You are now ready to finalize the creation of your account by:</p>

                <ul>
                    <li>Naming your account</li>
                    <li>Set a password, you will use to sign all required actions with the logion application when required. We strongly recommend you save your password with a password manager such as <a href="https://keepassxc.org/" target="_blank" rel="noreferrer">KeePassXC</a></li>
                    <li>Once done, click on the “Add the account with the generated seed “ orange button</li>
                </ul>

                <img src={`${process.env.PUBLIC_URL}/assets/landing/logion-step2.gif`} style={{maxWidth: "100%"}} alt="step 2 animation"/>

                <h2>D - logion application first access</h2>

                <p>Your account is now created, you are ready to use logion:</p>

                <ul>
                    <li>Close the Polkadot extension tab</li>
                    <li>Close the Firefox/Chrome extension store tab</li>
                    <li>On the logion tab, you should now see the authentification screen (if not, just reload the page)</li>
                    <li>Select the account you just created, click on “log in” blue button: the polkadot extension popup should appear</li>
                    <li>Type your password, select the “Remember my password for the next 15 minutes” option and click on the “Sign the message” orange button</li>
                </ul>

                <p>You are ready to use the logion wallet and Decentralized Application (Dapp) !</p>

                <p>We strongly recommend setting your legal protection as advised by the related message.</p>

                <img src={`${process.env.PUBLIC_URL}/assets/landing/logion-step3.gif`} style={{maxWidth: "100%"}} alt="step 3 animation"/>
            </>
        );
    } else {
        return (
            <>
                <p>Your browser is currently not supported, please try with
                <a href="https://www.google.com/chrome/">Chrome</a> or
                <a href="https://www.mozilla.org/firefox/">Firefox</a></p>
            </>
        );
    }
}
