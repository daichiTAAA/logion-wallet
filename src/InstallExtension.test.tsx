jest.mock('./logion-chain');

import React from 'react';
import InstallExtension from './InstallExtension';
import { shallowRender } from './tests';
import { setRecommendedExtension } from './logion-chain';

test('Given unsupported browser, when rendering, then show message', () => {
    setRecommendedExtension(null);
    const tree = shallowRender(<InstallExtension />);
    expect(tree).toMatchSnapshot();
});

test('Given Firefox, when rendering, then show link', () => {
    setRecommendedExtension({
        browser: "firefox",
        url: "firefox url"
    });
    const tree = shallowRender(<InstallExtension />);
    expect(tree).toMatchSnapshot();
});

test('Given Chrome, when rendering, then show link', () => {
    setRecommendedExtension({
        browser: "chrome",
        url: "chrome url"
    });
    const tree = shallowRender(<InstallExtension />);
    expect(tree).toMatchSnapshot();
});
