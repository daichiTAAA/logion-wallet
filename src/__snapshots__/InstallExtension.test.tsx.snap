// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given Chrome, when rendering, then show link 1`] = `
<Shell>
  <Jumbotron
    fluid={false}
  >
    <h1>
      Install the extension
    </h1>
  </Jumbotron>
  <p>
    Please install 
    <a
      href="chrome url"
    >
      this browser extension
    </a>
     in order to store your accounts in a secure way. Once done, authorize the Logion wallet to access it and create or import an account.
  </p>
</Shell>
`;

exports[`Given Firefox, when rendering, then show link 1`] = `
<Shell>
  <Jumbotron
    fluid={false}
  >
    <h1>
      Install the extension
    </h1>
  </Jumbotron>
  <p>
    Please install 
    <a
      href="firefox url"
    >
      this browser extension
    </a>
     in order to store your accounts in a secure way. Once done, authorize the Logion wallet to access it and create or import an account.
  </p>
</Shell>
`;

exports[`Given unsupported browser, when rendering, then show message 1`] = `
<Shell>
  <p>
    Your browser is currently not supported, please try with
    <a
      href="https://www.google.com/chrome/"
    >
      Chrome
    </a>
     or
    <a
      href="https://www.mozilla.org/firefox/"
    >
      Firefox
    </a>
  </p>
</Shell>
`;
